python/semantic_kernel/agents/chat_completion_agent.py

```diff
@@ -0,0 +1,210 @@
+
+from semantic_kernel.agents.agent_channel import AgentChannel
+from semantic_kernel.connectors.ai.chat_completion_client_base import ChatCompletionClientBase
+from semantic_kernel.connectors.ai.prompt_execution_settings import PromptExecutionSettings
+from semantic_kernel.const import DEFAULT_SERVICE_NAME
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.streaming_chat_message_content import StreamingChatMessageContent
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.exceptions import KernelServiceNotFoundError
+from semantic_kernel.functions.kernel_arguments import KernelArguments
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+if TYPE_CHECKING:
+    from semantic_kernel.kernel import Kernel
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+@experimental_class
+class ChatCompletionAgent(Agent):
+    """A KernelAgent specialization based on ChatCompletionClientBase.
+
+    Note: enable `function_choice_behavior` on the PromptExecutionSettings to enable function
+    choice behavior which allows the kernel to utilize plugins and functions registered in
+    the kernel.
+    """
+
+    service_id: str
+    execution_settings: PromptExecutionSettings | None = None
+    channel_type: ClassVar[type[AgentChannel]] = ChatHistoryChannel
+
+    def __init__(
+        self,
+        service_id: str | None = None,
+        kernel: "Kernel | None" = None,
+        name: str | None = None,
+        id: str | None = None,
+        description: str | None = None,
+        instructions: str | None = None,
+        execution_settings: PromptExecutionSettings | None = None,
+    ) -> None:
+        """Initialize a new instance of ChatCompletionAgent.
+
+        Args:
+            service_id: The service id for the chat completion service. (optional) If not provided,
+                the default service name `default` will be used.
+            kernel: The kernel instance. (optional)
+            name: The name of the agent. (optional)
+            id: The unique identifier for the agent. (optional) If not provided,
+                a unique GUID will be generated.
+            description: The description of the agent. (optional)
+            instructions: The instructions for the agent. (optional)
+            execution_settings: The execution settings for the agent. (optional)
+        """
+        if not service_id:
+            service_id = DEFAULT_SERVICE_NAME
+
+        args = {
+            "service_id": service_id,
+            "name": name,
+            "description": description,
+            "instructions": instructions,
+            "execution_settings": execution_settings,
+        }
+        if id is not None:
+            args["id"] = id
+        if kernel is not None:
+            args["kernel"] = kernel  # type: ignore
```

Comment:
If you specify dict[str, Any] when you create the args, then this type ignore can go away